python/lsst/analysis/tools/interfaces.py

```diff
@@ -420,6 +420,9 @@ def getOutputNames(self) -> Iterable[str]:
         """
         outNames = tuple(self.produce.getOutputNames())
         if outNames:
-            return outNames
+            return (f"{self.identity or ''}_{name}" for name in outNames)
```

Comment:
If `self.identity` is `None`, then the return here will be `_{name}`, which I don't think was the desired intention. This could be changed to be smarter about naming things, but I'm not going to push for it on this ticket.